File: Source/JavaScript/model/layouts/FlowGrid.ts
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

import { FlowContainer } from './FlowContainer';
import { FlowContainerKind } from './FlowContainerKind';

/**
 * A {@link FlowContainer} that arranges its children in a grid.
 */
export interface FlowGrid extends FlowContainer {
    kind: FlowContainerKind.Grid;
    columns?: number;
    rows?: number;
}

export const FlowGridPropertyNames: (keyof FlowGrid)[] = ['kind', 'columns', 'rows'];

File: Source/DotNET/Model/Layouts/FlowGrid.cs
// Copyright (c) Cratis. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Cratis.Scene.Model.Layouts;

/// <summary>
/// A <see cref="FlowContainer"/> that arranges its children in a grid.
/// </summary>
public record FlowGrid : FlowContainer
{
    /// <summary>
    /// Gets the number of columns, or <see langword="null"/> to size columns from content.
    /// </summary>
    public int? Columns { get; init; }

    /// <summary>
    /// Gets the number of rows, or <see langword="null"/> to size rows from content.
    /// </summary>
    public int? Rows { get; init; }
}
